nextcord/role.py

```diff
@@ -449,15 +450,9 @@ async def edit(
             payload['mentionable'] = mentionable
 
         if icon is not MISSING:
-            if icon is None:
-                payload['icon'] = icon
-            elif isinstance(icon, str):
+            if icon is None or isinstance(icon, str):
```

Comment:
Don't need to set unicode emoji if icon is `None`

```suggestion
            if isinstance(icon, str):
```